File: server/src/RSVP.Core/Models/Invite.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RSVP.Core.Models
{
    public class Invite
    {
        public Guid InviteId { get; set; }
        public required string InviteName { get; set; }
        public List<Guest>? Guests { get; set; }
        public DateTime CreatedDateTime { get; set; }
    }
}

File: client/src/types/Invite.ts
import type { Guest, CreateGuestPayload } from "./Guest";

export interface Invite {
    inviteId: string;
    inviteName: string;
    guests: Guest[] | null;
}

export interface CreateInvitePayload {
    inviteName: string;
    guests: CreateGuestPayload[] | null;
    guestIds: string[] | null;
}

export interface UpdateInvitePayload {
    inviteId: string;
    inviteName: string;
    guests: CreateGuestPayload[] | null;
    guestIds: string[] | null;
}